src/nbl/asset/CShaderIntrospector.cpp

```diff
@@ -698,16 +698,18 @@ static void introspectStructType(spirv_cross::Compiler& _comp, impl::SShaderMemo
                 member.count_specID = sc->id;
             }
         }
-		else if (mtype.basetype != spirv_cross::SPIRType::Struct) // might have to ignore a few more types than structs
-			member.arrayStride = core::max(0x1u<<core::findMSB(member.size),16u); // @Crisspl  NOT TRUE FOR MATRICES!
+        else if (mtype.columns > 1u)
+            member.arrayStride = member.size;
+		else if (mtype.basetype != spirv_cross::SPIRType::Struct)
+			member.arrayStride = core::max(0x1u<<core::findMSB(member.size),16u);
```

Comment:
explain your reasoning?

Before all types except for structs got their strides aligned up to PoT, the only problem was that non square matrices (or even a `mat2`) might have a size bigger than 16 bytes and their array stride could be out of whack.

Now you've disabled this stride determination for everything but matrices (no vectors) and the same issue as above remains for matrices.